09_json/alextmz/pkg/puppy/types_test.go

```diff
@@ -0,0 +1,59 @@
+package puppy
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMarshall(t *testing.T) {
+	var tests typetest
+	buildTestvar(&tests)
+
+	for name, tt := range tests {
+		test := tt
+		t.Run(name, func(t *testing.T) {
+			got, err := json.Marshal(test.obj)
+			assert.NoError(t, err)
+			assert.JSONEq(t, test.jsn, string(got))
```

Comment:
About the `json.Marshall` and `Unmarshal`, see above. 
For the JSONstr, I made it a lambda instead - having a custom formatter allows me to keep all the fixtures in one place instead of duplicating them all over. Is this bad?